scipy/stats/tests/test_stats.py

```diff
@@ -768,79 +768,88 @@ def test_gh_8111(self):
         assert_allclose([res1, res2, res3], expected)
 
 
-def test_spearmanr():
-    # Cross-check with R:
-    # cor.test(c(1,2,3,4,5),c(5,6,7,8,7),method="spearmanr")
-    x1 = [1, 2, 3, 4, 5]
-    x2 = [5, 6, 7, 8, 7]
-    expected = (0.82078268166812329, 0.088587005313543798)
-    res = stats.spearmanr(x1, x2)
-    assert_approx_equal(res[0], expected[0])
-    assert_approx_equal(res[1], expected[1])
-
-    attributes = ('correlation', 'pvalue')
-    res = stats.spearmanr(x1, x2)
-    check_named_results(res, attributes)
-
-    # with only ties in one or both inputs
-    with np.errstate(invalid="ignore"):
-        assert_equal(stats.spearmanr([2,2,2], [2,2,2]), (np.nan, np.nan))
-        assert_equal(stats.spearmanr([2,0,2], [2,2,2]), (np.nan, np.nan))
-        assert_equal(stats.spearmanr([2,2,2], [2,0,2]), (np.nan, np.nan))
-
-    # empty arrays provided as input
-    assert_equal(stats.spearmanr([], []), (np.nan, np.nan))
-
-    np.random.seed(7546)
-    x = np.array([np.random.normal(loc=1, scale=1, size=500),
-                np.random.normal(loc=1, scale=1, size=500)])
-    corr = [[1.0, 0.3],
-            [0.3, 1.0]]
-    x = np.dot(np.linalg.cholesky(corr), x)
-    expected = (0.28659685838743354, 6.579862219051161e-11)
-    res = stats.spearmanr(x[0], x[1])
-    assert_approx_equal(res[0], expected[0])
-    assert_approx_equal(res[1], expected[1])
-
-    assert_approx_equal(stats.spearmanr([1,1,2], [1,1,2])[0], 1.0)
-
-    # test nan_policy
-    x = np.arange(10.)
-    x[9] = np.nan
-    assert_array_equal(stats.spearmanr(x, x), (np.nan, np.nan))
-    assert_allclose(stats.spearmanr(x, x, nan_policy='omit'),
-                    (1.0, 0))
-    assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='raise')
-    assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='foobar')
+class TestCorrSpearmanr2(object):
```

Comment:
why r2?

Reply:
> why r2?

Mostly because I lack creativity in naming pytest classes. And there was already this one:
https://github.com/scipy/scipy/blob/47ffc1ef2480df307392670a5476701b57e5333c/scipy/stats/tests/test_stats.py#L543
which looked like it duplicated most of the tests, that is why I deleted that single one you spotted beneath. The (linked) test class is already really big so I didn't think it would be good  
to make it bigger so I created a new one. Open to better name suggestions or if there is substantial duplication (I think there is) also open to removing as much as is reasonable.

Reply:
Ok, I will take a look at the tests in the existing class

Reply:
I had a quick look at the existing class. There is already a test that checks the attribute names, so you can indeed remove test_spearmanr_attributes from the new (2) class. Otherwise let's leave the tests as they are

Reply:
ok, removed.